spec/models/user_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { create :user }
+  
+  it { should validate_presence_of :username }
+  it { should validate_presence_of :email }
+  it { should validate_uniqueness_of :username }
+  it { should validate_uniqueness_of :email }
+  
```

Comment:
check out adding [`it { should have_secure_password }`](http://matchers.shoulda.io/docs/v2.8.0/Shoulda/Matchers/ActiveModel.html#have_secure_password-instance_method)